### ISSUE TITLE
compile issue with electron 12

### ISSUE THREAD

**Issue by Strit** [bitwarden/jslib#306]
When building jslib (as part of bitwraden) I get these errors, which I assume is because we switched to electron 12:

```
[Main] [24] ./src/main.ts + 17 modules 86.7 KiB {0} [built] [1 warning]
[Main]      | ./src/main.ts 7.8 KiB [built]
[Main]      | ./src/services/i18n.service.ts 979 bytes [built]
[Main]      | ./src/main/menu.main.ts 20.7 KiB [built]
[Main]      | ./src/main/messaging.main.ts 5.21 KiB [built]
[Main]      | ./src/main/powerMonitor.main.ts 3.15 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/keytarStorageListener.ts 1.77 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/services/electronLog.service.ts 1.19 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/services/electronMainMessaging.service.ts 444 bytes [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/services/electronStorage.service.ts 769 bytes [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/tray.main.ts 6.31 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/updater.main.ts 6.42 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/window.main.ts 11.9 KiB [built] [1 warning]
[Main]      | ./src/main/nativeMessaging.main.ts 9.2 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/utils.ts 1.07 KiB [built]
[Main]      | ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/baseMenu.ts 6.46 KiB [built]
[Main]      |     + 3 hidden modules
[Main] [25] ./src/entry.ts + 3 modules 6.04 KiB {0} [built]
[Main]      | ./src/entry.ts 1.01 KiB [built]
[Main]      | ./src/proxy/native-messaging-proxy.ts 580 bytes [built]
[Main]      | ./src/proxy/ipc.ts 1.45 KiB [built]
[Main]      | ./src/proxy/nativemessage.ts 3 KiB [built]
[Main]     + 11 hidden modules
[Main] 
[Main] WARNING in ../jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/window.main.ts
[Main] Module Warning (from ./node_modules/tslint-loader/index.js):
[Main] [121, 17]: members are not aligned
[Main] [122, 17]: members are not aligned
[Main] [123, 17]: members are not aligned
[Main] [124, 17]: members are not aligned
[Main] 
[Main]  @ ./src/main.ts 24:0-56 77:30-40
[Main]  @ ./src/entry.ts
[Main] 
[Main] ERROR in /build/bitwarden/src/desktop-1.25.0/jslib/src/electron/window.main.ts
[Main] [tsl] ERROR in /build/bitwarden/src/desktop-1.25.0/jslib/src/electron/window.main.ts(124,17)
[Main]       TS1117: An object literal cannot have multiple properties with the same name in strict mode.
[Main] 
[Main] ERROR in /build/bitwarden/src/desktop-1.25.0/jslib/src/electron/window.main.ts
[Main] [tsl] ERROR in /build/bitwarden/src/desktop-1.25.0/jslib/src/electron/window.main.ts(124,17)
[Main]       TS2300: Duplicate identifier 'enableRemoteModule'.
[Main] 
[Main] ERROR in /build/bitwarden/src/jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/window.main.ts
[Main] [tsl] ERROR in /build/bitwarden/src/jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/window.main.ts(124,17)
[Main]       TS1117: An object literal cannot have multiple properties with the same name in strict mode.
[Main] 
[Main] ERROR in /build/bitwarden/src/jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/window.main.ts
[Main] [tsl] ERROR in /build/bitwarden/src/jslib-f80e89465ffc004705d2941301c0ffb6bfd71d1a/src/electron/window.main.ts(124,17)
[Main]       TS2300: Duplicate identifier 'enableRemoteModule'.
[Main] npm ERR! code ELIFECYCLE
npm ERR! errno 2
[Main] npm ERR! bitwarden@0.0.0 build:main: `webpack --config webpack.main.js`
[Main] npm ERR! Exit status 2
[Main] npm ERR! 
npm ERR! Failed at the bitwarden@0.0.0 build:main script.
[Main] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[Main] 
[Main] npm ERR! A complete log of this run can be found in:
[Main] npm ERR!     /build/bitwarden/src/npm_cache/_logs/2021-03-12T15_34_38_876Z-debug.log
[Main] npm run build:main exited with code 2
[Rend] Hash: 1e603c4acffdd16e94cf
[Rend] Version: webpack 4.29.0
[Rend] Time: 59421ms
[Rend] Built at: 03/12/2021 3:35:04 PM
[Rend]  2 assets
[Rend] Entrypoint app/main = app/main.js
[Rend] [0] ./src/app/main.ts 0 bytes {0} [built]
[Rend] 
[Rend] ERROR in jslib/src/electron/window.main.ts:124:17 - error TS1117: An object literal cannot have multiple properties with the same name in strict mode.
[Rend] 
[Rend] 124                 enableRemoteModule: true, // TODO: This needs to be removed prior to Electron 14.
[Rend]                     ~~~~~~~~~~~~~~~~~~
[Rend] jslib/src/electron/window.main.ts:124:17 - error TS2300: Duplicate identifier 'enableRemoteModule'.
[Rend] 
[Rend] 124                 enableRemoteModule: true, // TODO: This needs to be removed prior to Electron 14.
[Rend]                     ~~~~~~~~~~~~~~~~~~
[Rend] 
[Rend] Child html-webpack-plugin for "index.html":
[Rend]      1 asset
[Rend]     Entrypoint undefined = index.html
[Rend]     [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 600 bytes {0} [built]
[Rend] npm ERR! code ELIFECYCLE
[Rend] npm ERR! errno 2
[Rend] npm ERR! bitwarden@0.0.0 build:renderer: `gulp prebuild:renderer && webpack --config webpack.renderer.js`
[Rend] npm ERR! Exit status 2
[Rend] npm ERR! 
[Rend] npm ERR! Failed at the bitwarden@0.0.0 build:renderer script.
[Rend] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[Rend] 
[Rend] npm ERR! A complete log of this run can be found in:
[Rend] npm ERR!     /build/bitwarden/src/npm_cache/_logs/2021-03-12T15_35_05_400Z-debug.log
[Rend] npm run build:renderer exited with code 2
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bitwarden@0.0.0 build: `concurrently -n Main,Rend -c yellow,cyan "npm run build:main" "npm run build:renderer"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bitwarden@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /build/bitwarden/src/npm_cache/_logs/2021-03-12T15_35_05_549Z-debug.log
==> ERROR: A failure occurred in build().
```

**Comment by Strit** [comment]
Seems it's not just electron 12 that has this issue.

Also happens with Electron 11.

Build system is Manjaro ARM (aarch64). Was able to build it previously. Last successful build was in late January, first failed build was in early March. So something happened between there.

**Comment by Hinton** [comment]
@Strit I seem to remember seeing similar errors before, I believe they occur when trying to build desktop if you have a `node_modules` directory in the jslib submodule. Could you try deleting the `node_modules` directories and do a clean `npm install` inside the desktop project?

**Comment by Strit** [comment]
Thanks for the response.

I always build in clean build folders, and we use the jslib tarball, not the submodule in bitwarden.

I did however find that the above error was because of a change I did in my build environment, where I add the enableRemoteModule manually. But since you guys now do that, it failed. But now I get a new error relating to electron folder location. But it's related to the desktop application, not jslib anymore as far as I can see.
